managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/DimensionStats.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2016 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.mledger.util;
+
+import org.HdrHistogram.Histogram;
+import org.HdrHistogram.Recorder;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ */
+public class DimensionStats {
+
+    /** Statistics for given dimension **/
+    public double meanDimensionMs;
+
+    public double medianDimensionMs;
+
+    public double dimension95Ms;
+
+    public double dimension99Ms;
+
+    public double dimension999Ms;
+
+    public double dimension9999Ms;
+
+    public double dimensionCounts;
+
+    public double elapsedIntervalMs;
+
+    private Recorder dimensionTimeRecorder = new Recorder(TimeUnit.MINUTES.toMillis(10), 2);
+    private Histogram dimensionHistogram = null;
+
+    public void updateStats() {
+
+        dimensionHistogram = dimensionTimeRecorder.getIntervalHistogram(dimensionHistogram);
+
+        this.meanDimensionMs = dimensionHistogram.getMean();
+        this.medianDimensionMs = dimensionHistogram.getValueAtPercentile(50);
+        this.dimension95Ms = dimensionHistogram.getValueAtPercentile(95);
+        this.dimension99Ms = dimensionHistogram.getValueAtPercentile(99);
+        this.dimension999Ms = dimensionHistogram.getValueAtPercentile(99.9);
+        this.dimension9999Ms = dimensionHistogram.getValueAtPercentile(99.99);
+        this.dimensionCounts = dimensionHistogram.getTotalCount();
+    }
+
+    public void recordValue(long dimensionLatencyMs) {
+        dimensionTimeRecorder.recordValue(dimensionLatencyMs);
```

Comment:
Actually [new Recorder(int,int)](https://github.com/rdhabalia/pulsar/blob/b39f318c998face61b7c5c28731bd37951aef283/managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/DimensionStats.java#L44)  returns [AtomicHistogram](https://hdrhistogram.github.io/HdrHistogram/JavaDoc/org/HdrHistogram/AtomicHistogram.html) and as per documentation is a thread-safe. However, [new Recorder(int)](https://github.com/HdrHistogram/HdrHistogram/blob/master/src/main/java/org/HdrHistogram/Recorder.java) creates [ConcurrentHistogram](https://hdrhistogram.github.io/HdrHistogram/JavaDoc/org/HdrHistogram/ConcurrentHistogram.html). So, do you think we should use `ConcurrentHistogram` which is also thread-safe. ??